Modules/Microsoft.PowerShell.UnixCompleters/OnStart.ps1

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Informative messages to let users know that completers have been registered
+
+Write-Verbose "Registering UNIX native util completers"
+
+# We don't have access to the module at load time, since loading occurs last
+# Instead we set up a one-time event to set the OnRemove scriptblock once the module has been loaded
+$null = Register-EngineEvent -SourceIdentifier PowerShell.OnIdle -MaxTriggerCount 1 -Action {
+    $m = Get-Module PSUnixUtilCompleters
+    $m.OnRemove = {
+        Write-Verbose "Deregistering UNIX native util completers"
```

Comment:
This is better moved to a psm1 file, where you can use `$MyInvocation.MyCommand.Module.OnRemove`.

Reply:
After actually trying this out, it doesn't seem to work: `$MyInvocation.MyCommand.Module` is null, I suspect because the psm1 being run is still in the dot-source phase; it hasn't been created as a module yet. That's why I had to be tricky with that event

Reply:
Try `$MyInvocation.MyCommand.ScriptBlock.Module.OnRemove` instead. This one should work.